File: data structures and algorithms/java/06_020.java
import java.io.File;
import java.nio.file.Files;
import java.nio.file.Path;

/**
 * Recursively explore directory structure
 */
public void exploreDirectory(String path, int depth) {
   // Validate directory path
   if (!Files.isDirectory(Path.of(path))) {
      return;
   }
   
   // Display current directory with indentation
   System.out.println("  ".repeat(depth) + 
      new File(path).getName());
   
   // Explore subdirectories using DFS approach
   try {
      for (File item : new File(path).listFiles()) {
         String fullPath = item.getPath();
         exploreDirectory(fullPath, depth + 1);
      }
   } catch (Exception e) {
      System.err.println("Error accessing directory: " + 
         e.getMessage());
   }
}

File: data structures and algorithms/python/06_020.py
def explore_directory(path, depth=0):
    if not os.path.isdir(path):
        return
        
    # Process current directory
    print('  ' * depth + os.path.basename(path))
    
    # Explore subdirectories (DFS!)
    for item in os.listdir(path):
        full_path = os.path.join(path, item)
        explore_directory(full_path, depth + 1)